python/ql/lib/semmle/python/regexp/internal/RegExpTracking.qll

```diff
@@ -0,0 +1,78 @@
+/**
+ * Provides predicates that track strings to where they are used as regular expressions.
+ * This is implemented using TypeTracking in two phases:
+ *
+ * 1: An exploratory backwards analysis that imprecisely tracks all nodes that may be used as regular expressions.
+ * The exploratory phase ends with a forwards analysis from string constants that were reached by the backwards analysis.
+ * This is similar to the exploratory phase of the JavaScript global DataFlow library.
+ *
+ * 2: A precise type tracking analysis that tracks constant strings to where they are used as regular expressions.
+ * This phase keeps track of which strings and regular expressions end up in which places.
+ */
```

Comment:
Yes. At least it did when I wrote the same thing for Ruby.  

Your simple solution will have more tuples because the `regExpBacktracker` predicate could track many paths that never reach any string-constants.  
And I the number of tuples is low in the exploratory phase because it doesn't keep track of which source/sink flow came from.  

It's the same basic idea in the first phase of the global dataflow library (at least in JavaScript, I assume the shared dataflow library does some of the same).